python/cudf/cudf/core/column/column.py

```diff
@@ -1212,6 +1212,45 @@ def _with_type_metadata(self: ColumnBase, dtype: Dtype) -> ColumnBase:
         """
         return self
 
+    def _label_encoding(self, cats, dtype=None, na_sentinel=-1):
+        from cudf._lib.join import join as cpp_join
+
+        def _return_sentinel_column():
+            return cudf.core.column.full(
+                size=len(self), fill_value=na_sentinel, dtype=dtype
+            )
+
+        if dtype is None:
+            dtype = min_scalar_type(max(len(cats), na_sentinel), 8)
+
+        cats = as_column(cats)
```

Comment:
This is an internal method. Do we not already know at this point whether `cats` is a column or not?

Reply:
I made it so that we always pass a column.